bin/flutter

```diff
@@ -35,8 +35,9 @@ function path_uri() {
 
 function upgrade_flutter () {
   if hash flock 2>/dev/null; then
```

Comment:
I don't understand the trickery going on here.
Can you add comments explaining this?
What does `hash flock` do? Why are we locking fd 3?

Reply:
Added some comments, PTAL. :-)